react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      username: '',
+      password: ''
+    };
+
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({
+      password: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    console.log(`state: ${this.state}`);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>Username:
+        <input type="text" value={this.state.username} onChange={this.handleUsernameChange}></input>
+        </label>
+        <label>
+          Password:
+          <input type="password" value={this.state.password} onChange={this.handlePasswordChange}></input>
+        </label>
+        <input type="submit" value="Sign up" />
+      </form>
+    );
+  }
+}
+
+const root = document.querySelector('#root');
+ReactDOM.render(
+  <RegistrationForm />,
+  root
+);
```

Comment:
This is the old way of using `render` and will not be supported much longer, you should be using the current way:
```JSX
const container = document.querySelector('#root');
const root = ReactDOM.createRoot(container);

root.render(<RegistrationForm />);
```